### PR TITLE
Implement B/X style gems and Jewelry generation

### PR DESCRIPTION
**Pull request by ShockTohp**
Gems and Jewelry are implemented. No breaking changes.

### CODE REVIEW

File: src/app/generate/services/generate-treasure/generate-bx-treasure/bx-gems/bx-gems.service.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      quantity: 1,
+      value: this.generateGemValue()
+    } as ValueablesResult)
+    gem.name = `Gem (${gem.value} ${gem.denomination})`;
+    return gem;
+  }
+
+  generateGemValue(): number {
+    const roll = rollDice(this.d20)
+    return this.GEM_ROLL_TABLE[roll - 1] //Zero indexed Array
```
Comment (TaylorLearnsToCode) [inline src/app/generate/services/generate-treasure/generate-bx-treasure/bx-gems/bx-gems.service.ts:59]: As arrays should be! ( ;-) )

**Comment by TaylorLearnsToCode** [comment]
Looking good so far! For the gem/jewel service - when it generates the ValuablesResult collection, there is a chance it will generate multiple results with the same values: e.g. - 
> [ { value: 10, quantity: 1 }, { value: 10, quantity: 1 } ]

...instead of -
> [ { value: 10, quantity: 2 } ]

I recommend checking out the LBB services - [Gem](https://github.com/ShockTohp/greg/blob/feature/support-bx/src/app/generate/services/generate-treasure/generate-lbb-treasure/lbb-gems/lbb-gems.service.ts#L69-L78) or [Jewelry](https://github.com/ShockTohp/greg/blob/feature/support-bx/src/app/generate/services/generate-treasure/generate-lbb-treasure/lbb-jewels/lbb-jewels.service.ts#L71-L80): they have a method, "`addOrIncrement*`" which will combine them into a more concise list. You could copy/paste, if you wanted, or - better, as now that I'm looking at it, it looks like I did some copy pasting - extract it into an exported function in a new utility file (maybe, `treasure.util.ts`) in the Shared module.

File: src/app/generate/services/generate-treasure/generate-bx-treasure/bx-jewelry/bx-jewelry.service.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        this.jewelryResult.push(jewelry)
+      }
+
+      this.jewelryResult.sort((a, b) => a.value - b.value);
+      return this.jewelryResult;
+    }
+    return null;
+  }
+
+  generateJewelry(): ValueablesResult {
```
Comment (TaylorLearnsToCode) [inline src/app/generate/services/generate-treasure/generate-bx-treasure/bx-jewelry/bx-jewelry.service.ts:33]: Methods unreferenced outside the class in which they are defined should probably be marked as "private"

**Review by TaylorLearnsToCode — APPROVED** [review]

File: src/app/generate/services/generate-treasure/generate-bx-treasure/bx-jewelry/bx-jewelry.service.ts (hunk 2 above)
Comment (ShockTohp) [reply src/app/generate/services/generate-treasure/generate-bx-treasure/bx-jewelry/bx-jewelry.service.ts:33]: Remediated, didn't realize that was a best practice for node. PR incoming.

**Comment by ShockTohp** [comment]
> Looking good so far! For the gem/jewel service - when it generates the ValuablesResult collection, there is a chance it will generate multiple results with the same values: e.g. -
> 
> > [ { value: 10, quantity: 1 }, { value: 10, quantity: 1 } ]
> 
> ...instead of -
> 
> > [ { value: 10, quantity: 2 } ]
> 
> I recommend checking out the LBB services - [Gem](https://github.com/ShockTohp/greg/blob/feature/support-bx/src/app/generate/services/generate-treasure/generate-lbb-treasure/lbb-gems/lbb-gems.service.ts#L69-L78) or [Jewelry](https://github.com/ShockTohp/greg/blob/feature/support-bx/src/app/generate/services/generate-treasure/generate-lbb-treasure/lbb-jewels/lbb-jewels.service.ts#L71-L80): they have a method, "`addOrIncrement*`" which will combine them into a more concise list. You could copy/paste, if you wanted, or - better, as now that I'm looking at it, it looks like I did some copy pasting - extract it into an exported function in a new utility file (maybe, `treasure.util.ts`) in the Shared module.

Moving this into an issue and opening a branch for it separately. There will likely end up being a slew of utilities that need to be added as more systems are implemented. I'll label it as maintenance/utility-creation or similiar.

**Pull request merged**